installers/osie/main.go

```diff
@@ -10,99 +10,138 @@ import (
 	"go.opentelemetry.io/otel/trace"
 )
 
-type Installer struct {
-	// ExtraKernelArgs are key=value pairs to be added as kernel commandline to the kernel in iPXE for OSIE.
-	ExtraKernelArgs string
+type installer struct {
+	osieURL string
+
+	defaultParams string
+	tinkParams    string
+	hollowParams  string
 }
 
-// DefaultHandler determines the ipxe boot script to be returned.
-// If the job has an instance with Rescue true, the rescue boot script is returned.
-// Otherwise the installation boot script is returned.
-func (i Installer) DefaultHandler() job.BootScript {
-	return func(ctx context.Context, j job.Job, s ipxe.Script) ipxe.Script {
-		if j.Rescue() {
-			return i.rescue()(ctx, j, s)
+// Installer instantiates a new osie installer.
+func Installer(dataModelVersion, tinkGRPCAuth, extraKernelArgs, registry, registryUsername, registryPassword string, tinkTLS bool) job.BootScripter {
+	defaultParams := []string{
+		"ip=dhcp",
+		"modules=loop,squashfs,sd-mod,usb-storage",
+		"alpine_repo=${base-url}/repo-${arch}/main",
+		"modloop=${base-url}/modloop-${parch}",
+		"tinkerbell=${tinkerbell}",
+		"syslog_host=${syslog_host}",
+		"parch=${parch}",
+		"packet_action=${action}",
+		"packet_state=${state}",
+		"osie_vendors_url=" + conf.OsieVendorServicesURL,
+	}
+
+	if extraKernelArgs != "" {
+		defaultParams = append(defaultParams, extraKernelArgs)
+	}
+
+	i := installer{
+		osieURL:       conf.MirrorBaseURL + "/misc/osie",
+		defaultParams: strings.Join(defaultParams, " "),
+	}
+
+	if conf.HollowClientId != "" && conf.HollowClientRequestSecret != "" {
+		hollowParams := []string{
+			"hollow_client_id=" + conf.HollowClientId,
+			"hollow_client_request_secret=" + conf.HollowClientRequestSecret,
 		}
+		i.hollowParams = strings.Join(hollowParams, " ")
+	}
 
-		return i.install()(ctx, j, s)
+	if dataModelVersion != "1" {
+		return i
 	}
-}
 
-// rescue generates the ipxe boot script for booting into osie in rescue mode
-func (i Installer) rescue() job.BootScript {
-	return func(ctx context.Context, j job.Job, s ipxe.Script) ipxe.Script {
-		s.Set("action", "rescue")
-		s.Set("state", j.HardwareState())
+	tinkParams := []string{
+		"grpc_authority=" + tinkGRPCAuth,
+		"packet_base_url=" + conf.MirrorBaseURL + "/workflow",
+	}
+	if !tinkTLS {
+		tinkParams = append(tinkParams, "tinkerbell_tls=false")
+	}
+	if registry != "" {
+		tinkParams = append(tinkParams, "docker_registry="+registry)
+	}
+	if registryUsername != "" {
+		tinkParams = append(tinkParams, "registry_username="+registryUsername)
+	}
+	if registryPassword != "" {
+		tinkParams = append(tinkParams, "registry_password="+registryPassword)
+	}
```

Comment:
I believe these would still be needed even if `dataModelVersion != "1"`

Reply:
thats not how the code is today: https://github.com/tinkerbell/boots/blob/main/installers/osie/main.go#L123-L139. This is behind `CanWorkflow` but that only ever happens when dataModelVersion == 1 anyway. We shouldn't have 2 ways to do this check.

Reply:
thats not my experience. i can use the `DATA_MODEL_VERSION=standalone` and still use `CanWorkflow`.

Reply:
hmm this seems very accidental and confusing, if you're in dmv=standalone and canworkflow=true then why would tink connection params matter? dmv is actually doing 2 different jobs and causing confusion (surprise, surprise... :smile:). `DATA_SOURCE=cacher|tink|standalone|something-in-the-future/pluggable` DATA_MODEL=cacher|tink`. But oh well, not going to try and do that now. I'll fix it so tinkArgs isn't dependent on DMV=1.

Reply:
done